05-html-css/flexbox-intro.md

```diff
@@ -0,0 +1,230 @@
+
+## Example
+
+Take a moment to read through [this example on Codepen](https://codepen.io/adadev/pen/jzNQmV?editors=1100) that we'll use for the rest of the lecture.
+
+What's in there? What is it currently doing? Why are the elements that size?
+
+Now add the `display: flex;` rule to the `.container` rule:
+```css
+.container {
+  display: flex;
+}
+```
+
+Our elements used to take up the whole width of their parent container because by default, `div`s take their size from their default `display: block;`. However, once the elements became _flex items_ (due to being direct children of a flex container with `display: flex;`), each element stopped taking up an entire "row."
+
+## Controlling Flow
+
+Flexbox defines some specific vocabulary that applies to flow.
+
+In flexbox, there is a **main axis**. This axis runs **along** the direction that inline items flow.
+
+![Main Axis: row](./imgs/flexbox-axis-main-row.png)
+
+![Main Axis: column](./imgs/flexbox-axis-main-column.png)
+
+There is also the **cross axis**. This axis runs **perpendicular** to the main axis.
+
+![Cross Axis: row](./imgs/flexbox-axis-cross-row.png)
+
+![Cross Axis: column](./imgs/flexbox-axis-cross-column.png)
+
+You can determine the direction of the main axis with the property `flex-direction`.
+
+`flex-direction` has the following properties:
+- `row` (default)
+- `row-reverse`
+- `column`
+- `column-reverse`
+
+**Practice:** Pair up with someone around you and make a prediction for what each of these will do. Then, add the property `flex-direction` to [the Codepen](https://codepen.io/adadev/pen/jzNQmV?editors=1100) and change the values and observe the differences.
+
+## Determining Wrapping Behavior
+
+We can see in our example right now that our flex items extend beyond the flex container. Let's change that: we can determine the behavior for what our flex items should do if it fills the flex container along the main axis using `flex-wrap` on the flex container:
+
+```css
+.container {
+  display: flex;
+  flex-wrap: wrap;
+}
+```
+
+The available values for `flex-wrap` are:
+- `nowrap`
+- `wrap`
+- `wrap-reverse`
+
+**Practice:** Using [the same Codepen](https://codepen.io/adadev/pen/jzNQmV?editors=1100), add the property `flex-wrap` and each of the values to observe the differences.
+
+## Controlling Alignment
+
+Let's look at a modification of the example we've been following. [Look at this Codepen](https://codepen.io/adadev/pen/ZxzVLq?editors=1100); it's nearly identical, but we've added some stuff that essentially shows a more realistic case where every flex item is not identical in width or height. This will help us see empty space between flex items better.
+
+![Varied Width and Height Items, unaligned](./imgs/flexbox-alignment-example-1.png)
+
+How are the flex items aligned against each other right now? Where is there empty space/negative space? Why is it there? What kind of alignment would be more pleasant to look at/be a better experience for a website?
+
```

Comment:
Can you present these questions a little more formally? Something like:

**Questions:**
- How are the flex items currently aligned?
- Where is there empty space / negative space?
  - Where did the negative space come from?
- What kind of an alignment would be more pleasant to look at/be a better experience for a website?